src/api/mail/controller.js

```diff
@@ -0,0 +1,39 @@
+import { sendJson } from 'utils/api';
+import config from 'config';
+import HttpError from 'node-http-error';
+import HttpStatus from 'http-status-codes';
+
+const Mailchimp = require('mailchimp-api-v3');
+const md5 = require('md5');
+
+export const getEmailAddressHash = emailAddress => md5(emailAddress.trim().toLowerCase());
+
+export const updateEmailAddressStatus = (req, res, next) => {
+  const { apiKey, listId } = config.mail.mailchimp;
+  const mailchimp = new Mailchimp(apiKey);
+  const { emailAddress, userStatus, language } = req.body;
+
+  try {
+    if (!(emailAddress && userStatus && language))
```

Comment:
Here we in fact perform validation of the request. It would be clearer if you implement a middleware for that. Example - `setMainPageValidator`: https://github.com/babajka/babajka-backend/blob/develop/src/utils/validation.js#L82
Then you can export it as a `precheck`: https://github.com/babajka/babajka-backend/blob/develop/src/utils/validation.js#L87
And use it in your `index.js` file when you define your API call: https://github.com/babajka/babajka-backend/blob/develop/src/api/storage/index.js#L17